crates/subspace-runtime/src/lib.rs

```diff
@@ -746,6 +747,25 @@ fn extract_bundles(extrinsics: Vec<OpaqueExtrinsic>) -> Vec<OpaqueBundle> {
         .collect()
 }
 
+fn extrinsics_shuffling_seed<Block: BlockT>(header: Block::Header) -> Randomness {
+    let pre_digest = if header.number().is_zero() {
+        PreDigest::<FarmerPublicKey>::genesis_pre_digest()
+    } else {
+        let mut pre_digest: Option<_> = None;
+        for log in header.digest().logs() {
+            match (log.as_subspace_pre_digest(), pre_digest.is_some()) {
+                (Some(_), true) => panic!("Multiple Subspace pre-runtime digests in a header"),
+                (None, _) => {}
+                (s, false) => pre_digest = s,
+            }
+        }
+
+        pre_digest.expect("Header must contain one pre-runtime digest; qed")
+    };
+
+    BlakeTwo256::hash_of(&pre_digest.solution.signature).into()
+}
```

Comment:
This looks cleaner and doesn't require additional impl in `sp-consensus-subspace`:
```suggestion
fn extrinsics_shuffling_seed<Block: BlockT>(header: Block::Header) -> Randomness {
    if header.number().is_zero() {
        Randomness::default()
    } else {
        let mut pre_digest: Option<_> = None;
        for log in header.digest().logs() {
            match (log.as_subspace_pre_digest(), pre_digest.is_some()) {
                (Some(_), true) => panic!("Multiple Subspace pre-runtime digests in a header"),
                (None, _) => {}
                (s, false) => pre_digest = s,
            }
        }

        let pre_digest = pre_digest.expect("Header must contain one pre-runtime digest; qed")
        BlakeTwo256::hash_of(&pre_digest.solution.signature).into()
    }
}
```

Reply:
Done.